bookkeeper-server/src/main/java/org/apache/bookkeeper/util/DiskChecker.java

```diff
@@ -155,6 +156,29 @@ float checkDiskFull(File dir) throws DiskOutOfSpaceException, DiskWarnThresholdE
     }
 
     /**
+     * calculates and returns the disk usage factor in the provided list of dirs
+     * 
+     * @param dirs
+     *            list of directories
+     * @return disk usage factor in the provided list of dirs
+     */
+    public float getTotalDiskUsage(List<File> dirs) {
+        if (dirs == null || dirs.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "list argument of getTotalDiskUsage is not supposed to be null or empty");
+        }
+        long totalUsableSpace = 0;
+        long totalSpace = 0;
+        for (File dir : dirs) {
+            totalUsableSpace += dir.getUsableSpace();
+            totalSpace += dir.getTotalSpace();
```

Comment:
valid point! internally we made assumption that ledger/index dirs wont be on the same partition. but current implementation doesn't restrict user from doing this. I think it needs little more than warn message. Logged Feature story - https://issues.apache.org/jira/browse/BOOKKEEPER-1033 . Will take this issue separately, since it needs to be handled in couple of other places as well.